mm2src/coins/z_coin.rs

```diff
@@ -1023,27 +995,50 @@ impl<'a> ZCoinBuilder<'a> {
             // This is not used for Zcoin so we just provide a default value
             path_to_address: HDPathAccountToAddressId::default(),
         };
-        ZCoinBuilder {
+
+        let z_spending_key = match z_spending_key {
+            Some(ref z_spending_key) => z_spending_key.clone(),
+            None => extended_spending_key_from_protocol_info_and_policy(
+                &protocol_info,
+                &priv_key_policy,
+                z_coin_params.account,
+            )?,
+        };
+
+        let (_, my_z_addr) = z_spending_key
+            .default_address()
+            .map_err(|_| MmError::new(ZCoinBuildError::GetAddressError))?;
```

Comment:
I think we should refactor all such `map_err` conversions where the target is MmError<E> into `map_to_mm`:
this will track location and looks more neatly

Reply:
i see this doesn't affect location tracing tho. a new mmerror is still initialized ([see](https://github.com/KomodoPlatform/komodo-defi-framework/blob/4e11c571a22163c75c2a37cd8bcb98ee32ef27ba/mm2src/mm2_err_handle/src/map_to_mm.rs#L38)).

Reply:
[1741f34](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/1741f340a61400774c627ed85442faa5eb7ce5f7)

Reply:
yes does not affect location indeed, only looks more mm2-idiomatic :-)